### PR TITLE
UI refinement

### PR DESCRIPTION
**Pull request by tkohr**
PR intents to refine the UI implementation based on present mockups.

localhost_4200_metadata_ee965118-2416-4d48-b07e-bbc696f002c2 (1)

### CODE REVIEW

**Comment by fgravin** [comment]
Good job
Last things to fix 
- font in `mat-table`
- Link section: put button in secondary when hover
- Handle the related metadata link click (it should change the route instead of opening the md in gn3)

File: libs/ui/elements/src/lib/related-record-card/related-record-card.component.html [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
-      [href]="record.metadataUrl"
-      target="_blank"
-      class="cursor-pointer"
-      mat-raised-button
-      [matTooltip]="'tooltip.url.open' | translate"
-      matTooltipPosition="above"
-    >
-      <mat-icon class="align-middle text-secondary">open_in_new</mat-icon>
-    </a>
+    <div class="absolute" style="bottom: 1rem">
```
Comment (fgravin) [inline libs/ui/elements/src/lib/related-record-card/related-record-card.component.html:19]: no, should be done with the flex system

**Review by fgravin — CHANGES_REQUESTED** [review]
good work 2 small comments
also we are think with @jahow to create a `title` component that would just display the title with the correct classes and sizes, to avoid adding classes in `styles.css` file.

File: apps/datahub/src/app/app.module.ts [hunk 2]
```diff
@@ -100,5 +100,10 @@ export class AppModule {
       getThemeConfig().PRIMARY_COLOR,
       [25]
     )
+    ThemeService.generateBgOpacityClasses(
```
[the remark this answers is not in the document]
Comment (tkohr) [reply apps/datahub/src/app/app.module.ts]: oups, must have been tired

File: libs/ui/elements/src/lib/related-record-card/related-record-card.component.html (hunk 1 above)
Comment (tkohr) [reply libs/ui/elements/src/lib/related-record-card/related-record-card.component.html]: adapted the three cards accordingly

**Comment by tkohr** [comment]
Thanks for the feedback!

For the routing, I'm wondering what's the best architectural choice, here. Using `RouterFacade` directly inside `RelatedCardComponent` creates a circular dependency, and even using it inside `FeatureRecordModule` does.

Regarding the custom CSS classes I was wondering how we could avoid just that, defining classes in datahub, and using them in the libs. Do I understand correctly that you propose to define them in the new "TitleComponent"? Even then, we'll need to indicate which class to use when using the component, right? Btw. I called them `text-title-...`, but finally they are also used in some other places, so we might choose a different name for the component.

**Comment by tkohr** [comment]
So the expected routing has been added with a `routerLink` and custom CSS classes have been moved from the global style sheet to the tailwind config. Some other adaptions we've discussed have also been made. If you don't see any hicks, I think this PR should be ready to merge.

**Comment by fgravin** [comment]
Are you sure you catched all usage of `title-font` for the size ?
Like the metadata results list preview title is not in your changes ..
Also seems to miss some sections in the record view, `text-28` is just applied 3 times while present in a lot in the pages.

**Comment by fgravin** [comment]
The go to metadata route should scroll the body to the top.
While navigating to related records, we are scrolled down, same when we come from the result list

**Comment by fgravin** [comment]
I search for "scot", enter a metadata, navitage to related records back and fourth, then an error in the console appears sometimes
```
ERROR TypeError: Cannot read properties of undefined (reading 'protocol')
    at LinkHelperService.isWfsLink (link-helper.service.ts:66)
    at DataViewTableComponent.fetchData (data-view-table.component.ts:86)
    at SwitchMapSubscriber.project (data-view-table.component.ts:66)
    at SwitchMapSubscriber._next (switchMap.js:28)
    at SwitchMapSubscriber.next (Subscriber.js:49)
    at MapSubscriber._next (map.js:35)
    at MapSubscriber.next (Subscriber.js:49)
    at CombineLatestSubscriber.notifyNext (combineLatest.js:73)
    at InnerSubscriber._next (InnerSubscriber.js:11)
    at InnerSubscriber.next (Subscriber.js:49)
```

for instance moving from `ee965118-2416-4d48-b07e-bbc696f002c2`
to `urn:isogeo:metadata:uuid:0a83d03d-e95d-469a-9323-f4f9045ed1d8`

**Comment by fgravin** [comment]
Links card layout is not sharp enough
Screenshot from 2021-12-13 18-17-57

I don't know if the title should go on 2 lines or be ellipsed (TW classes exist for both cases), but they should not but cropped this way.
Also, if you cut the title, we should be able to see it on mouse hover

**Comment by fgravin** [comment]
http://localhost:4200/metadata/emp_act.pt_pxx_fact1564

the downloads card also has a weird behavior
it should be sized

**Comment by fgravin** [comment]
The Preview active tab border color should be primary
```css
.mat-tab-group.mat-primary .mat-ink-bar, .mat-tab-nav-bar.mat-primary .mat-ink-bar {
    background-color: #134895;
}
```

**Comment by tkohr** [comment]
> great ! No need to change the `lineHeight` for other sizes ?

They are all 'normal' in the mockups, besides the 48 that I've also added now.

**Comment by tkohr** [comment]
> Are you sure you catched all usage of `title-font` for the size ? Like the metadata results list preview title is not in your changes .. Also seems to miss some sections in the record view, `text-28` is just applied 3 times while present in a lot in the pages.

Ah, good eyes! Hope I got all with 88cebe8c5fb0eb92032f31faeb2d115d9d5aceec now

**Comment by tkohr** [comment]
> The go to metadata route should scroll the body to the top. While navigating to related records, we are scrolled down, same when we come from the result list

Added e2ae6f592b569ea741ef0f667b875fc0466f195b for that

**Comment by tkohr** [comment]
> Links card layout is not sharp enough Screenshot from 2021-12-13 18-17-57
> 
> I don't know if the title should go on 2 lines or be ellipsed (TW classes exist for both cases), but they should not but cropped this way. Also, if you cut the title, we should be able to see it on mouse hover

Hum, did you get the screenshots from this branch ? The text is wrapped for same md on my side...

**Comment by tkohr** [comment]
> http://localhost:4200/metadata/emp_act.pt_pxx_fact1564
> 
> the downloads card also has a weird behavior it should be sized

This bc98295d97f2579ffa3678715074b93f761d585f should prevent weird overflows

**Comment by tkohr** [comment]
> The Preview active tab border color should be primary
> 
> ```css
> .mat-tab-group.mat-primary .mat-ink-bar, .mat-tab-nav-bar.mat-primary .mat-ink-bar {
>     background-color: #134895;
> }
> ```

Ok, added primary color for that edaeef8696b7843b186b68eaec60c346815e63ab

**Comment by tkohr** [comment]
> I search for "scot", enter a metadata, navitage to related records back and fourth, then an error in the console appears sometimes
> 
> ```
> ERROR TypeError: Cannot read properties of undefined (reading 'protocol')
>     at LinkHelperService.isWfsLink (link-helper.service.ts:66)
>     at DataViewTableComponent.fetchData (data-view-table.component.ts:86)
>     at SwitchMapSubscriber.project (data-view-table.component.ts:66)
>     at SwitchMapSubscriber._next (switchMap.js:28)
>     at SwitchMapSubscriber.next (Subscriber.js:49)
>     at MapSubscriber._next (map.js:35)
>     at MapSubscriber.next (Subscriber.js:49)
>     at CombineLatestSubscriber.notifyNext (combineLatest.js:73)
>     at InnerSubscriber._next (InnerSubscriber.js:11)
>     at InnerSubscriber.next (Subscriber.js:49)
> ```
> 
> for instance moving from `ee965118-2416-4d48-b07e-bbc696f002c2` to `urn:isogeo:metadata:uuid:0a83d03d-e95d-469a-9323-f4f9045ed1d8`

This bug occurred, when navigating from a record with table data to a record without table data. In this case, the values of     `MdViewFacade.dataLinks$` and `MdViewFacade.geoDataLinks$` seem to be updated in `data-view-table.component` before `record-metadata.component` which results in undefined link values, that were not handled in `data-view-table.component`. 125710d41f78925680bc605ed05e5968b78f915a should resolve this.

**Pull request merged**